### PR TITLE
Add regex property to one of required option.

### PR DESCRIPTION
**Pull request by alxlnk**
#20211

### CODE REVIEW

**Comment by samdark** [comment]
Would you please add a line for changelog and a test?

File: tests/framework/widgets/ActiveFieldTest.php [hunk 1]
```diff
@@ -691,6 +692,16 @@ public function testInputOptionsTransferToWidget()
             'options' => ['placeholder' => 'pholder_both_direct']
         ]);
         $this->assertStringContainsString('placeholder="pholder_both_direct"', (string) $widget);
+
+        try {
+            $widget = $this->activeField->widget(TestMaskedInput::className(), [
```
Comment (mtangoo) [inline tests/framework/widgets/ActiveFieldTest.php:697]: Why is there no assertion to test when no exception is thrown? Why also did you add try...catch?

**Review by mtangoo — CHANGES_REQUESTED** [review]

File: tests/framework/widgets/ActiveFieldTest.php (hunk 1 above)
Comment (alxlnk) [reply tests/framework/widgets/ActiveFieldTest.php:697]: Perhaps you can tell me what exactly needs to be tested in this case? After all, previously the initialization of the widget was not covered by the test.

Also, I have never written tests before)

Comment (mtangoo) [reply tests/framework/widgets/ActiveFieldTest.php:697]: You are basically testing the case where  you configure only regex things should go fine. So it is similar to previous tests (see example before) only different configurations are being tested

 ```
        $widget = $this->activeField->widget(TestMaskedInput::className(), [
            'mask' => '999-999-9999',
            'options' => ['placeholder' => 'pholder_direct'],
        ]);
        $this->assertStringContainsString('placeholder="pholder_direct"', (string) $widget);
```

Comment (mtangoo) [reply tests/framework/widgets/ActiveFieldTest.php:697]: You do not need to try/catch as exception will cause the test to fail anyway

**Review by mtangoo — APPROVED** [review]

**Comment by terabytesoftw** [comment]
You could try the exception and the message, by using regex on the widget without the option alias, and wait.

```php
$this->expectException('yii\base\InvalidConfigException');
$this->expectExceptionMessage('$this message exception')
```

**Review by terabytesoftw — APPROVED** [review]

**Comment by mtangoo** [comment]
@alxlnk thanks for your time and efforts

**Pull request merged**